File: external/webroot/md/codeExample/Face-API-Identify/FaceIdentify.py
# encoding:utf-8
import base64
import urllib
import urllib2

from token import access_token

'''
人脸查找之识别接口
'''

identifyUrl = "https://aip.baidubce.com/rest/2.0/face/v2/identify"
# 参数group_id：用户组id（由数字、字母、下划线组成）images：图像base64编码,多张图片半角逗号分隔
params = {"group_id": "gid",
          "images": "【图片base64编码】"}
params = urllib.urlencode(params)
access_token = access_token.AuthService()
identifyUrl = identifyUrl + "?access_token=" + access_token
request = urllib2.Request(url=identifyUrl, data=params)
request.add_header('Content-Type', 'application/x-www-form-urlencoded')
response = urllib2.urlopen(request)
content = response.read()
if content:
    print content

File: external/webroot/md/codeExample/Face-API-Identify/FaceIdentify.cs
using System;
using System.IO;
using System.Net;
using System.Text;
using System.Web;

namespace com.baidu.ai
{
	public class Identify
	{
		// 人脸识别
		public static String identify()
		{
			string token = "########";
			string base645 = FileUtils.getFileBase64("/work/ai/images/face/search5.jpg");
			string host = "https://aip.baidubce.com/rest/2.0/face/v2/identify?access_token=" + token;
			Encoding encoding = Encoding.Default;
			HttpWebRequest request = (HttpWebRequest)WebRequest.Create(host);
			request.Method = "post";
			request.ContentType = "application/x-www-form-urlencoded";
			request.KeepAlive = true;
			String str = "images=" + HttpUtility.UrlEncode(base645);
			str += "&group_id=demo&user_top_num=1&face_top_num=1";
			byte[] buffer = encoding.GetBytes(str);
			request.ContentLength = buffer.Length;
			request.GetRequestStream().Write(buffer, 0, buffer.Length);
			HttpWebResponse response = (HttpWebResponse)request.GetResponse();
			StreamReader reader = new StreamReader(response.GetResponseStream(), Encoding.Default);
			string result = reader.ReadToEnd();
			Console.WriteLine("人脸识别:");
			Console.WriteLine(result);
			return result;

		}
	}
}
